crates/gateway-client/src/lib.rs

```diff
@@ -103,6 +103,118 @@ pub trait GatewayApi: Sync {
     }
 }
 
+#[async_trait::async_trait]
```

Comment:
Wow. I'm surprised we need so much boilerplate for this...

Reply:
It took me a while to figure out as well :/ 

Rust is missing delegation / forwarding which makes this painful. Probably a macro out there that does this, but I'd rather just get rid of this entirely in the future.